Corona-Warn-App/src/main/java/de/rki/coronawarnapp/covidcertificate/validationrules/rules/DgcRulesValidator.kt

```diff
@@ -0,0 +1,34 @@
+package de.rki.coronawarnapp.covidcertificate.validationrules.rules
+
+import de.rki.coronawarnapp.covidcertificate.common.repository.CertificateContainerId
+import de.rki.coronawarnapp.covidcertificate.test.core.TestCertificateRepository
+import de.rki.coronawarnapp.covidcertificate.validationrules.country.DgcCountry
+import org.joda.time.DateTime
+import javax.inject.Inject
+
+class DgcRulesValidator @Inject constructor(
+    testCertificateRepository: TestCertificateRepository,
+    dgcDgcValidationRulesRepository: DgcValidationRulesRepository,
+) {
+
+    /**
+     * Validates DGC against country of arrival's rules and issuer country rules
+     */
+    suspend fun validateDgc(
+        arrivalsCountry: Set<DgcCountry>, // For future allow multiple country selection
+        containerId: CertificateContainerId,
+        dateTime: DateTime,
+        cwt: CWT
+    ): DgcValidationResult {
+        return object : DgcValidationResult {
+            override val expirationCheckPassed: Boolean
+                get() = false
+            override val jsonSchemaCheckPassed: Boolean
```

Comment:
Not sure about the sensibility here - qr codes with invalid json won't even get here. They are rejected when scanning/ retrieved if invalid.

Reply:
I agree, see discussions on internal teams channel.
Let's keep it like this for now and refactor after the discussion has concluded.